### PR TITLE
Stackdriver-based Trace API implementation

### PR DESCRIPTION
**Pull request by dmdashenkov**
This PR introduces the implementation of the new [Spine Trace API](https://github.com/SpineEventEngine/core-java/pull/1090) based on [Stackdriver Trace](https://cloud.google.com/trace/docs/).

When a command/an event is handled, a new span is created. Spans which represent the propagation of a single signal tree are grouped into a single trace.
For example, if the command `StartGroupImport` cased one event `GroupImportStarted` and many events `MemberImported`, some of which in turn caused several `UserCreated`, all of those commands and events belong to the same signal tree (with the root in `StartGroupImport`) and thus are displayed on the same trace.

The trace below shows an aggregate part `TaskPart` handle a `CreateTask` command and a `TaskViewProjection` handle the resulting `TaskCreated` event.

Screen Shot 2019-06-27 at 15 07 40

Spans include such information as the IDs of the signal and the receiver entity, the name of the bounded context to which the entity belongs, and the associated tenant ID.

The screenshot below shows the detailed info about a single span.
Screen Shot 2019-06-27 at 15 07 58

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: stackdriver-trace/src/main/java/io/spine/server/trace/stackdriver/SignalSpan.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import java.util.stream.Stream;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.protobuf.AnyPacker.unpack;
+import static java.lang.String.format;
+
+/**
+ * A span based on a signal.
+ *
+ * <p>Signal messages may be processed once or many times by different entities. Each such
```
Comment (armiol) [inline stackdriver-trace/src/main/java/io/spine/server/trace/stackdriver/SignalSpan.java]: Can you please tell more on the subject. I also don't think that `each such` is a good thing.

**Review by armiol — COMMENTED** [review]
@dmdashenkov LGTM with the comment.

**Comment by armiol** [comment]
@dmdashenkov also, I am not sure it's a good idea to have the span names that long. In particular, why the project ID is always included? Let's discuss vocally.

This PR description lacks a screenshot of the Google Console. Otherwise, we could see if the long names are OK.

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

**Comment by dmdashenkov** [comment]
The AppVeyor build [is not yet configured](#115), thus it fails.

File: stackdriver-trace/src/main/java/io/spine/server/trace/stackdriver/SignalSpan.java [hunk 2]
```diff
@@ -42,20 +43,36 @@
...
 
     private static final int SPAN_DISPLAY_NAME_LENGTH = 128;
+
     private final BoundedContextName context;
     private final Signal<?, ?, ?> signal;
     private final MessageId receiver;
+    private final EntityTypeName receiverType;
 
-    SignalSpan(BoundedContextName context, Signal<?, ?, ?> signal, MessageId receiver) {
+    protected SignalSpan(BoundedContextName context,
```
Comment (armiol) [inline stackdriver-trace/src/main/java/io/spine/server/trace/stackdriver/SignalSpan.java:76]: Please document.

File: stackdriver-trace/src/main/java/io/spine/server/trace/stackdriver/TraceService.java [hunk 3]
```diff
@@ -28,7 +28,15 @@
  * <p>Depending on the implementation, the client may send queries in the same thread or
  * asynchronously.
  */
-interface TraceService extends AutoCloseable {
+public interface TraceService extends AutoCloseable {
 
+    /**
+     * Performs the given {@code BatchWriteSpansRequest}.
```
Comment (armiol) [inline stackdriver-trace/src/main/java/io/spine/server/trace/stackdriver/TraceService.java]: I think it's `Executes`.

File: stackdriver-trace/src/main/java/io/spine/server/trace/stackdriver/SignalSpan.java [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        private EntityTypeName receiverType;
+
+        /**
+         * Prevents direct instantiation.
+         */
+        private Builder() {
+        }
+
+        public Builder setContext(BoundedContextName context) {
+            this.context = checkNotNull(context);
```
Comment (armiol) [inline stackdriver-trace/src/main/java/io/spine/server/trace/stackdriver/SignalSpan.java:189]: We don't seem to test these for `null` values.

**Review by armiol — COMMENTED** [review]
@dmdashenkov please see my comments. LGTM otherwise.

**Comment by dmdashenkov** [comment]
@armiol PTAL again.

**Review by armiol — APPROVED** [review]

**Pull request merged**